### DISCUSSION TITLE
features compared to GitHub

### DISCUSSION THREAD

**Discussion by ghost** [framesurge/perseus#220] [General]
Does perseus have enough features to write an alternative to GitHub?

**Comment by lukechu10** [comment]
Depends on what you are looking for. Please provide more details because we can't really help you with such a vague question.

Are you looking for a specific feature that you need?

**Reply by ghost** [reply]
Since GitHub is a code-hosting platform, it requires features to receive user data and store that data in engine-side (like a cloud disk), and requires account creation, since GitHub users must create an account to send code to GitHub. Perhaps, it requires CAPTCHA, since GitHub uses CAPTCHA to prevent spamming accounts.

**Reply by lukechu10** [reply]
> Since GitHub is a code-hosting platform, it requires features to receive user data and store that data in engine-side (like a cloud disk), and requires account creation, since GitHub users must create an account to send code to GitHub. Perhaps, it requires CAPTCHA, since GitHub uses CAPTCHA to prevent spamming accounts.

Those aren't really framework features. They're things you need to implement yourself, regardless of what framework you use...

**Reply by ghost** [reply]
Are those possible without using #[perseus::engine_main] and #[perseus::browser_main], i.e., just use #[perseus::main()]?
